crates/nu-parser/src/parser.rs

```diff
@@ -826,10 +826,26 @@ pub fn parse_internal_call(
             &signature,
             expand_aliases_denylist,
         );
+
         if let Some(long_name) = long_name {
             // We found a long flag, like --bar
-            error = error.or(err);
-            call.add_named((long_name, None, arg));
+            if matches!(err, Some(ParseError::UnknownFlag(_, _, _, _)))
+                && signature.allows_unknown_args
+            {
+                let (arg, arg_err) = parse_value(
+                    working_set,
+                    arg_span,
+                    &SyntaxShape::Any,
```

Comment:
Because we want to pass also other expressions that evaluate to a string, such as variables (`$foo`), subexpressions (`([a b c] | str collect ' ')`), etc. So what I did is I allowed all expressions and left the type checking up to the runtime.